### PR TITLE
82 publish abis and interfaces as a node module

### PR DESCRIPTION
**Pull request by katien**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -5,20 +5,7 @@
...
     "outDir": "dist",
-    "declaration": true,
-    "skipLibCheck": true,
-    "forceConsistentCasingInFileNames": true,
+    "declaration": true
   },
-  "allowJs": false,
-  "include": [
-    "./scripts",
-    "./test",
```
Comment (DarksightKellar) [inline tsconfig.json:15]: Was there a reason these config params were removed (ie `skipLibCheck`, `forceConsistentCasingInFileNames`, `"allowJs": false`, and `./test` in `include`)?

Or do they actually have no real use here, what with javascript type configuration being rubbish and all.

**Review by DarksightKellar — COMMENTED** [review]
Had a question on `tsconfig` changes, but other than that looks good to me!

File: tsconfig.json (hunk 1 above)
Comment (knoll3) [reply tsconfig.json:15]: I can't remember why these were added in here to be honest... Some of this looks like some old leftover stuff or possibly even junk copied over from fractal (haven't verified that).

**Review by knoll3 — APPROVED** [review]
Should be good after running eslint on some of these files.

File: tsconfig.json (hunk 1 above)
Comment (katien) [reply tsconfig.json:15]: These configurations were largely taken from fractal with a couple of minor changes to fit our project structure

**Review by sethhrbek — APPROVED** [review]

**Pull request merged**